blog-posts/typescript-ast-type-checker.md

```diff
@@ -0,0 +1,591 @@
+---
+path: "/blog/typescript-ast-type-checker"
+date: "2021-06-18"
+title: "Going beyond the Abstract Syntax Tree (AST) with the TypeScript Type Checker"
+shortSummary: How to analyze a typescript file beyond AST (Abstract Syntax Tree) with the tpe checker.
+author: Georgios Kaleadis
+authorSummary: "CTO at Satellytes"
+featuredImage: images/johann-siemens-EPy0gBJzzZU-unsplash.jpg
+attribution:
+    creator: Johann Siemens
+    source: https://unsplash.com/photos/EPy0gBJzzZU
+
+---
+
+For a recent Angular project we had to inspect a set of typescript files and output a table of involved types which are resolved down to their primitives. This post is about how to collect that data and why we need to go beyond the AST.
```

Comment:
I missing the "Common usecase" for this. Why is this important in general? I guess it's for automatic type generation for a component documentation?

Reply:
A small update to clarify the purpose. Very good hint as I should clarify the purpose of the whole drama a little bit better 😄

> In a recent Angular typescript project we defined data contracts between components based on generics. Later the components are arranged and connected within a low-code platform which will check if two components can work together based on their data contract. The checking process involves a json dataset of typing information which we collect by recursively resolving typescript types down to their primitives and then store the found property names along with their typing name. This post is about how to collect the typing information and why we need to go beyond the AST to achieve this.